client/data/settings/actions.js

```diff
@@ -13,6 +13,28 @@ import { __ } from '@wordpress/i18n';
 import ACTION_TYPES from './action-types';
 import { NAMESPACE, STORE_NAME } from '../constants';
 
+function setSettingsValue( payload ) {
+	return {
+		type: ACTION_TYPES.SET_SETTINGS_VALUE,
+		payload,
+	};
+}
+
+export function updateDigitalWalletsButtonActionType( type ) {
+	// eslint-disable-next-line camelcase
+	return setSettingsValue( { digital_wallets_button_action_type: type } );
+}
+
+export function updateDigitalWalletsButtonSize( size ) {
+	// eslint-disable-next-line camelcase
+	return setSettingsValue( { digital_wallets_button_size: size } );
+}
+
+export function updateDigitalWalletsButtonTheme( theme ) {
+	// eslint-disable-next-line camelcase
+	return setSettingsValue( { digital_wallets_button_theme: theme } );
+}
```

Comment:
We would reduce the number of functions here, but I think it wouldn't really simplify the logic in `client/settings/payment-method-settings/digital-wallets-settings.js`.

I was thinking that keeping them as separate functions helps if we'd like to move the UI components to separate components (instead of all controls being in one component). And maybe the hooks could even just use `setSettingsValue` directly instead.

But I'm guessing that would be overly abstracted, so I'm just happy to use whatever is easiest 😄

Reply:
> I was thinking that keeping them as separate functions helps if we'd like to move the UI components to separate components (instead of all controls being in one component).

Oh, I didn't think of that! I was thinking more of the storage and retrieval aspect of it. I guess even if they were managed by separate components, they could still only update their own partial state of a `DigitalWalletsAppearance` object? Or instead of using hooks, have a specific value that they control passed as a prop? 🤷

With the hook approach, a `useDigitalWalletsAppearance()` hook could return `[ style, updateStyle ]`, the latter of which would be a callback that'd accept a partial state (building on what I have learned about TS and partial states from your previous comment :D), and the `style` param could be destructured based on what the component actually needs.

Anyway, if you think it is better to have them decoupled on the front-end, maybe we could still remove the multiple methods on the back-end? Not changing anything here at all is also fine - I just noticed that there are bits of code that don't differ by much which I would probably try to reduce, and maybe wrongly so.

Reply:
Thanks for the thoughts!
I kept the separate hooks for the frontend for now, since I like that modular approach 👍 
I updated the backend in 69c1d6c8 to a single method. Hopefully the settings won't be refactored to a new page again, but I still tried to think of that with the `foreach`/`continue`. Let me know if you had something different in mind! :)